app/src/lib/databases/pull-request-database.ts

```diff
@@ -95,5 +96,162 @@ export class PullRequestDatabase extends BaseDatabase {
 
     // Remove the pullRequestStatus table
     this.conditionalVersion(5, { pullRequestStatus: null })
+
+    // Delete pullRequestsTable in order to recreate it again
+    // in version 7 with a new primary key
+    this.conditionalVersion(6, { pullRequests: null })
+
+    // new primary key and a new table dedicated to keeping track
+    // of the most recently updated PR we've seen.
+    this.conditionalVersion(7, {
+      pullRequests: '[base.repoId+number]',
+      pullRequestsLastUpdated: 'repoId',
+    })
+  }
+
+  /**
+   * Removes all the pull requests associated with the given repository
+   * from the database. Also clears the last updated date for that repository
+   * if it exists.
+   */
+  public async deleteAllPullRequestsInRepository(repository: GitHubRepository) {
+    const dbId = forceUnwrap(
+      "Can't delete PRs for repository, no dbId",
+      repository.dbID
+    )
+
+    await this.transaction(
+      'rw',
+      this.pullRequests,
+      this.pullRequestsLastUpdated,
+      async () => {
+        await this.clearLastUpdated(repository)
+        await this.pullRequests
+          .where('[base.repoId+number]')
+          .between([dbId], [dbId + 1])
+          .delete()
+      }
+    )
+  }
+
+  /**
+   * Create a pull request key from a GitHub repository and a PR number.
+   *
+   * This method is mainly a helper function to ensure we don't
+   * accidentally swap the order of the repository id and the pr number
+   * if we were to create the key array manually.
+   *
+   * @param repository The GitHub repository to which this PR belongs
+   * @param prNumber   The PR number as returned from the GitHub API
+   */
+  public getPullRequestKey(repository: GitHubRepository, prNumber: number) {
```

Comment:
:art: this could live outside the class as it doesn't depend on any internal state